(/authentication/social-connections/gitlab) update guide (#1672)
clerkinc/clerk-docs: docs/authentication/social-connections/gitlab.mdx

1. Select **Add connection** and select **For all users**.
1. In the **Choose provider** dropdown, select **GitLab**.
1. Select **Add connection**.

## Configure for your production instance

For _production instances_, you must provide custom credentials which involves generating your own **Client ID** and **Client Secret** using your GitLab account.

To make the setup process easier, it's recommended to keep two browser tabs open: one for your [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) and one for your [GitLab](https://gitlab.com) account.

<Steps>
  ### Enable GitLab as a social connection in Clerk

  1. In the Clerk Dashboard, navigate to the [**SSO Connections**](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) page.
  1. Select **Add connection** and select **For all users**.
  1. In the **Choose provider** dropdown, select **GitLab**.
  1. Ensure that both **Enable for sign-up and sign-in** and **Use custom credentials** are toggled on.
  1. Save the **Redirect URI** somewhere secure. Keep the modal and page open.

  ### Create a GitLab app

>   1. In Gitlab, navigate to the [**Applications**](https://gitlab.com/-/user_settings/applications) page.

Reviewer: ```suggestion
  1. In [Gitlab](https://gitlab.com), navigate to the [**Applications**](https://gitlab.com/-/user_settings/applications) page.
```

  1. Select **Add new application**.
  1. Enter your app name in **Name**.
  1. In **Redirect URI**, paste the **Redirect URI** you saved from the Clerk Dashboard.

>   1. Select the **Scopes** needed for your app. To allow users to sign up and sign in with GitLab, you must select `read_user` at a minimum.

Reviewer: This sounded awkward

```suggestion
  1. Select the **Scopes** needed for your app. To allow users to sign up and sign in with GitLab, you must at least select `read_user`.
```

>   1. Select **Save application**. You'll be redirected to your app's settings page. Save the **Application ID** and **Secret** somewhere secure. Keep this page open.

Reviewer: I don't think they have to keep the page open, they don't return to it.

```suggestion
  1. Select **Save application**. You'll be redirected to your app's settings page. Save the **Application ID** and **Secret** somewhere secure.
```

  ### Set the Client ID and Client Secret in your Clerk Dashboard

>   1. Navigate back to your Clerk Dashboard where the modal should still be open. Paste the **Application ID** in **Client ID** and **Secret** in **Client Secret**.

Reviewer: ```suggestion
  1. Navigate back to your Clerk Dashboard where the modal should still be open. Paste the **Application ID** and **Secret** values in the **Client ID** and **Client Secret** fields, respectively.
```
